plugins/metrics/epoch_commitments.go

```diff
@@ -0,0 +1,235 @@
+			transactionsKV,
+			objectstorage.LeakDetectionEnabled(false),
+			objectstorage.StoreOnCreation(true),
+		),
+	}
+	return metrics, nil
+}
+
+func (m *EpochCommitmentsMetrics) saveCommittedEpoch(ecr *epoch.ECRecord) {
+	m.epochCommitmentsMutex.Lock()
+	defer m.epochCommitmentsMutex.Unlock()
+	m.epochCommitments[ecr.EI()] = *ecr
+}
+
+func (m *EpochCommitmentsMetrics) GetCommittedEpochs() map[epoch.Index]epoch.ECRecord {
+	m.epochCommitmentsMutex.RLock()
+	defer m.epochCommitmentsMutex.RUnlock()
+	duplicate := make(map[epoch.Index]epoch.ECRecord, len(m.epochCommitments))
+	for k, v := range m.epochCommitments {
+		duplicate[k] = v
+	}
+	return duplicate
+}
+
+func (m *EpochCommitmentsMetrics) saveEpochVotersWeight(message *tangle.Message) {
+	branchesOfMessage, err := deps.Tangle.Booker.MessageBranchIDs(message.ID())
+	if err != nil {
+		Plugin.LogError(err)
+	}
+	voter := identity.NewID(message.IssuerPublicKey())
+	var totalWeight float64
+	_ = branchesOfMessage.ForEach(func(txID utxo.TransactionID) error {
+		weight := deps.Tangle.ApprovalWeightManager.WeightOfBranch(txID)
+		totalWeight += weight
+		return nil
+	})
+	m.epochVotersWeightMutex.Lock()
+	defer m.epochVotersWeightMutex.Unlock()
+
+	epochIndex := message.M.EI
+	if _, ok := m.epochVotersWeight[epochIndex]; !ok {
+		m.epochVotersWeight[epochIndex] = map[identity.ID]float64{}
+	}
+	m.epochVotersWeight[epochIndex][voter] += totalWeight
+}
+
+func (m *EpochCommitmentsMetrics) GetEpochVotersWeight() map[epoch.Index]map[identity.ID]float64 {
+	m.epochVotersWeightMutex.RLock()
+	defer m.epochVotersWeightMutex.RUnlock()
+	duplicate := make(map[epoch.Index]map[identity.ID]float64, len(m.epochVotersWeight))
+	for k, v := range m.epochVotersWeight {
+		subDuplicate := make(map[identity.ID]float64, len(v))
+		for subK, subV := range v {
+			subDuplicate[subK] = subV
+		}
+		duplicate[k] = subDuplicate
+	}
+	return duplicate
+}
+
+func (m *EpochCommitmentsMetrics) saveEpochUTXOs(tx *devnetvm.Transaction) {
+	earliestAttachment := deps.Tangle.MessageFactory.EarliestAttachment(utxo.NewTransactionIDs(tx.ID()))
+	ei := deps.NotarizationMgr.EpochManager().TimeToEI(earliestAttachment.IssuingTime())
+	createdOutputs := tx.Essence().Outputs().UTXOOutputs()
+	m.epochUTXOsMutex.Lock()
+	defer m.epochUTXOsMutex.Unlock()
+	obj := m.epochUTXOs.Load(ei.Bytes())
+	t := obj.Get()
+	if t == nil {
+		t = NewEpochContent(ei)
+	}
+	for _, o := range createdOutputs {
+		t.M.Content = append(t.M.Content, o.ID().String())
+	}
+	m.epochUTXOs.Store(t)
+}
+
+func (m *EpochCommitmentsMetrics) GetEpochUTXOs() map[epoch.Index][]string {
+	m.epochUTXOsMutex.RLock()
+	defer m.epochUTXOsMutex.RUnlock()
+	duplicate := make(map[epoch.Index][]string, 0)
+	m.epochUTXOs.ForEach(func(_ []byte, obj *objectstorage.CachedObject[*EpochContent]) bool {
+		ec := obj.Get()
+		duplicate[ec.M.EI] = ec.M.Content
+		return true
+	})
+	return duplicate
+}
+
+func (m *EpochCommitmentsMetrics) saveEpochMessage(msg *tangle.Message) {
```

Comment:
Same here. We want the exact data that the notarization manager used to create a certain commitment for an epoch.